File: 347 Top K Frequent Elements/347 Top K Frequent Elements.swift
class Solution {
  func topKFrequent(_ nums: [Int], _ k: Int) -> [Int] {
    let numsMap: [Int: Int] = nums.reduce(into: [:]) { map, num in map[num, default: 0] += 1 }

    return numsMap.sorted { $0.value > $1.value }.prefix(k).map { $0.key }
  }
}

File: 347 Top K Frequent Elements/347 Top K Frequent Elements.ts
function topKFrequent(nums: number[], k: number): number[] {
  const frequent = new Map<number, number>();

  for (const num of nums) {
    frequent.set(num, (frequent.get(num) || 0) + 1);
  }

  return [...frequent]
    .sort((a, b) => b[1] - a[1])
    .slice(0, k)
    .map((t) => t[0]);
}
